cmd/agent/common/helpers.go

```diff
@@ -53,8 +63,10 @@ func setupConfig(confFilePath string, configName string, withoutSecrets bool) (*
 	} else {
 		warnings, err = config.Load()
 	}
-	if err != nil {
-		return warnings, fmt.Errorf("unable to load Datadog config file: %s", err)
+	// If `!failOnMissingFile`, do not issue an error if we cannot find the default config file.
+	var e viper.ConfigFileNotFoundError
+	if err != nil && !(!failOnMissingFile && errors.As(err, &e) && confFilePath == "") {
```

Comment:
👍

Reply:
Done !
The `if` condition is now in normal conjonctive form. ;-)